### PR TITLE
Extend CPU Plugin with reference implementations of `StringTensorUnpack-15` and `StringTensorPack-15`

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Call reference implementations for `StringTensorPack/Unpack-15` operators
 - Add tests

### Tickets:
 - CVS-143930
 - CVS-143935

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@maxnick could you please review?

**Review by praasz — APPROVED** [review]

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#include "gtest/gtest.h"
+
+namespace ov {
+namespace test {
+namespace StringTensorPack {
+using StringTensorPackSpecificParams =  std::tuple<
+        InputShape,
+        InputShape,
+        std::vector<int64_t>,
+        std::vector<int64_t>
```
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.hpp:20]: Could you please add explanatory comments per each field?

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        InputShape indicesShape;
+        InputShape symbolsShape;
+        std::vector<int64_t> begins;
+        std::vector<int64_t> ends;
+        std::tie(indicesShape, symbolsShape, begins, ends) = StringTensorPackParams;
+
+        init_input_shapes({indicesShape, indicesShape, symbolsShape});
+        auto symbolsParameter = std::make_shared<ov::op::v0::Parameter>(ov::element::u8, inputDynamicShapes[2]);
+        auto beginsConst = std::make_shared<ov::op::v0::Constant>(indicesPrecision, indicesShape.second.front(), begins);
+        auto endsConst = std::make_shared<ov::op::v0::Constant>(indicesPrecision, indicesShape.second.front(), ends);
```
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.cpp:83]: Are these inputs supposed to be constants only in real applications?

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_unpack.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        InputShape{{-1}, {{3}}}
+    },
+    StringTensorUnpackSpecificParams {
+        InputShape{{-1, -1, -1}, {{1, 1, 13}}}
+    },
+    StringTensorUnpackSpecificParams {
+        InputShape{{-1, 10}, {{100, 10}}}
+    },
+    StringTensorUnpackSpecificParams {
+        InputShape{{-1, {7, 9}, 12, {1, 20}, -1}, {{3, 8, 12, 8, 4}}}
```
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_unpack.cpp:101]: Please add at least 2 different target shapes to test dynamic shapes change.

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.cpp [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        InputShape{{-1, -1, -1}, {{1, 3, 4}}},
+        InputShape{{-1}, {{108}}},
+        std::vector<int64_t>{7, 28, 30, 30, 41, 42, 50, 50, 74, 76, 80, 92},
+        std::vector<int64_t>{28, 30, 30, 41, 42, 50, 50, 74, 76, 80, 92, 108}
+    },
+    StringTensorPackSpecificParams {
+        InputShape{{-1, {1, 4}, {2, 3}}, {{1, 2, 2}}},
+        InputShape{{{50, 100}}, {{67}}},
+        std::vector<int64_t>{30, 31, 56, 60},
+        std::vector<int64_t>{31, 56, 60, 67}
```
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.cpp:124]: Please add at least 2 different target shapes to test dynamic shapes change.

**Review by maxnick — CHANGES_REQUESTED** [review]

File: src/plugins/intel_cpu/src/nodes/string_tensor_pack.cpp [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+namespace ov {
+namespace intel_cpu {
+namespace node {
+StringTensorPack::StringTensorPack(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr context)
+    : Node(op, context, NgraphShapeInferFactory(op, PortMask(0, 1))) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        OPENVINO_THROW_NOT_IMPLEMENTED(errorMessage);
+    }
+    const auto stringTensorPack = ov::as_type_ptr<const ov::op::v15::StringTensorPack>(op);
```
[the remark this answers is not in the document]
Comment (p-wysocki) [reply src/plugins/intel_cpu/src/nodes/string_tensor_pack.cpp]: Removed.

File: src/plugins/intel_cpu/src/nodes/string_tensor_unpack.cpp [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+struct StringTensorUnpackContext {
+    StringTensorUnpack &node;
+};
+}
+
+struct StringTensorUnpack::StringTensorUnpackExecute {
+    void operator()(StringTensorUnpackContext & ctx) {
+            ctx.node.executeImpl();
+        }
+};
```
[the remark this answers is not in the document]
Comment (p-wysocki) [reply src/plugins/intel_cpu/src/nodes/string_tensor_unpack.cpp]: Simplified.

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.cpp (hunk 2 above)
Comment (p-wysocki) [reply src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.cpp]: They're rather supposed to be Parameters, good that you noticed. I changed them to Parameters.

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_unpack.cpp (hunk 3 above)
Comment (p-wysocki) [reply src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_unpack.cpp]: Done.

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.cpp (hunk 4 above)
Comment (p-wysocki) [reply src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.cpp]: Done.

File: src/plugins/intel_cpu/src/nodes/string_tensor_unpack.cpp [hunk 7]
```diff
@@ -75,22 +74,13 @@ void StringTensorUnpack::executeImpl() {
...
-};
-}
-
 struct StringTensorUnpack::StringTensorUnpackExecute {
-    void operator()(StringTensorUnpackContext & ctx) {
-            ctx.node.executeImpl();
+    void operator()(StringTensorUnpack& node) {
+            node.executeImpl();
         }
 };
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/string_tensor_unpack.cpp:81]: Why can't we simply call `executeImpl()` inside the `execute` method? Moreover we can inline `executeImpl()` code inside the `execute` method as the `executeImpl()` is used once.

Comment (p-wysocki) [reply src/plugins/intel_cpu/src/nodes/string_tensor_unpack.cpp]: Good point, done.

**Review by mitruska — APPROVED** [review]

File: src/plugins/intel_cpu/src/nodes/string_tensor_unpack.h [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    void getSupportedDescriptors() override;
+    void initSupportedPrimitiveDescriptors() override;
+    void execute(dnnl::stream strm) override;
+    bool created() const override;
+    bool needPrepareParams() const override;
+    void executeDynamicImpl(dnnl::stream strm) override;
+
+private:
+    void executeImpl();
+    struct StringTensorUnpackExecute;
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/string_tensor_unpack.h:27]: Please clean up these signatures as not used.

Comment (p-wysocki) [reply src/plugins/intel_cpu/src/nodes/string_tensor_unpack.h]: Done.

**Review by maxnick — APPROVED** [review]

**Pull request merged**